microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java

```diff
@@ -278,12 +279,40 @@ private SeContainer createContainter(ClassLoader classLoader) {
                                                               .asOptionalMap()
                                                               .orElse(CollectionsHelper.mapOf()));
             initializer.setProperties(props);
+
+            // add resource classes explicitly configured without CDI annotations
+            this.resourceClasses.stream()
+                    .filter(this::notACdiBean)
+                    .forEach(initializer::addBeanClasses);
+
+
+            this.applications.stream()
+                    .map(JaxRsApplication::getConfig)
+                    .map(ResourceConfig::getClasses)
+                    .flatMap(Set::stream)
+                    .filter(this::notACdiBean)
+                    .forEach(initializer::addBeanClasses);
+
             STARTUP_LOGGER.finest("Initializer");
             SeContainer container = initializer.initialize();
             STARTUP_LOGGER.finest("Initalizer.initialize()");
             return container;
         }
 
+        private boolean notACdiBean(Class<?> clazz) {
+            if (clazz.getAnnotation(RequestScoped.class) != null) {
```

Comment:
`Dependent` too, right?

Also, what about custom scopes?  Those should be rejected as well, right?

Reply:
Is there a way to get a list of custom scopes before CDI initialization?

Reply:
No.  A portable extension, which runs as the container is initializing, may add any it likes.